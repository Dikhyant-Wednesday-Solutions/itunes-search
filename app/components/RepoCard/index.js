/**
 *
 * RepoCard
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card } from 'antd';
import T from '@components/T';

const CustomCard = styled(Card)`
  && {
    margin: 20px 0;
  }
`;

export function RepoCard({ name, fullName, stargazersCount }) {
  return (
    <CustomCard data-testid="repo-card">
      <T id="repository_name" values={{ name: name }} />
      <T id="repository_full_name" values={{ fullName: fullName }} />
      <T id="repository_stars" values={{ stars: stargazersCount }} />
    </CustomCard>
  );
}

RepoCard.propTypes = {
  name: PropTypes.string,
  fullName: PropTypes.string,
  stargazersCount: PropTypes.number
};

export default RepoCard;
