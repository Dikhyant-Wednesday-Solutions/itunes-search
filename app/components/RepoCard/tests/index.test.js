/**
 *
 * Tests for RepoCard
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import RepoCard from '../index';

describe('<RepoCard />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<RepoCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 RepoCard component', () => {
    const { getAllByTestId } = renderWithIntl(<RepoCard />);
    expect(getAllByTestId('repo-card').length).toBe(1);
  });
});
