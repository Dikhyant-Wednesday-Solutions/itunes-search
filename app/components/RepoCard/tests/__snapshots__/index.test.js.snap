// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered RepoCard__CustomCard-sc-3wf6pv-0 iiVuom"
      data-testid="repo-card"
    >
      <div
        class="ant-card-body"
      >
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="t"
        >
          Repository Name: 
        </p>
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="t"
        >
          Repository full name: 
        </p>
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="t"
        >
          Repository stars: 
        </p>
      </div>
    </div>
  </div>
</body>
`;
